### PR TITLE
Add pod annotationfor ASOv2 helm chart

### LINKED ISSUE #1367: Feature: Add pod annotations to aso helm charts
<Describe the feature request>

- the current helm charts don't support the ability to add pod annotations.
- ex: pod annotations are useful where we want our aso creds to be accessed from vault etc.

### PR DESCRIPTION
**Pull request by super-harsh**
Closes #1367 

**What this PR does / why we need it**:

This PR adds support for having pod annotations for ASOv2 helm chart. 

**Special notes for your reviewer**:

Didn't include new helm `.tgz` bundle in this PR, considering including it in the beta.1 release helm chart.

### CODE REVIEW

**Review by matthchr — APPROVED** [review]
LGTM. Agree not to include a new chart with this PR. We can make an updated chart for beta.1

File: v2/charts/azure-service-operator/values.yaml [hunk 1]
```diff
@@ -53,3 +53,6 @@ metrics:
 
 # installCRDs determines if the CRDs subchart should be installed.
 installCRDs: true
+
+#podAnnotations contain the pod annotations for Azure Service Operator
```
Comment (matthchr) [inline v2/charts/azure-service-operator/values.yaml:57]: minor: 
```suggestion
# podAnnotations contain the pod annotations for Azure Service Operator
```

**Comment by OleksandrBrodskiy** [comment]
Hi , seems like in this change you have an error with inserting data, ` {{ toYaml .Values.podAnnotations | indent 6 }}` , indent not needed here. Now it works like this : 
```
  template:
    metadata:
      annotations:
              testAnnotation: test
        kubectl.kubernetes.io/default-container: manager
```

**Comment by matthchr** [comment]
Thanks for the heads up @OleksandrBrodskiy, @super-harsh could you help take a look?

**Comment by super-harsh** [comment]
@OleksandrBrodskiy Thanks for raising this up. I'm getting a similar result, will send a PR shortly with the fix.

**Comment by OleksandrBrodskiy** [comment]
Thanks for the quick fix, is it possible to update the chart with this change?

**Comment by OleksandrBrodskiy** [comment]
And one more thing, for me it works only with this configuration :
```
        {{- if .Values.podAnnotations }}
        {{- toYaml .Values.podAnnotations| nindent 8 -}}
        {{- end }}
```
without indent, my annotation was also misplaced in the file, values example: 
```
podAnnotations:
  test.annotation: |
        [{
          "key": "value"
        }]
```

**Comment by super-harsh** [comment]
@OleksandrBrodskiy The change would be in the `beta.4` release. Which we're planning to publish soon. Also, for trying out your case, follow [helm documentation on multiline strings](https://helm.sh/docs/chart_template_guide/yaml_techniques/#strings-in-yaml).

**Pull request merged**